plugin/src/functionalTest/kotlin/io/qalipsis/gradle/cloud/CloudRunQalipsisTaskFunctionalTest.kt

```diff
@@ -0,0 +1,655 @@
+                import io.qalipsis.gradle.cloud.tasks.CloudRunQalipsis  
+                
+                plugins {
+                    id("io.qalipsis.cloud")
+                    java
+                }
+                tasks.register<CloudRunQalipsis>("qalipsisRunCampaign") {
+                    campaign {
+                        campaignName.set("Dummy campaign configuration")
+                        speedFactor.set(2.0)
+                        startOffsetMs.set(2000)
+                        campaignTimeout.set("PT1H30M")
+                        hardTimeout.set(false)
+                        scenario("My scenario 1") {
+                            minionsCount = 7
+                            zones {
+                               "CH" to 45
+                               "AS" to 55
+                            }
+                            profile {
+                                more(
+                                    periodMs = 11, 
+                                    minionsCountProLaunchAtStart = 23, 
+                                    multiplier = 2.0, 
+                                    maxMinionsCountProLaunch = 7
+                                )
+                            }
+                        }
+                    }
+                }
+            """.trimIndent()
+        )
+        mockServerClient.`when`(
+            request()
+                .withPath("/api/campaigns")
+                .withMethod("POST")
+                .withHeader(Header("Authorization", "Bearer wrong-token"))
+                .withBody(
+                    JsonBody.json(
+                        """
+                            {
+                            "name": "Dummy campaign configuration",
+                            "speedFactor": 2.0,
+                            "startOffsetMs": 2000,
+                            "timeout": "PT1H30M",
+                            "hardTimeout": false,
+                            "scenarios": {
+                                "My scenario 1": {
+                                    "minionsCount": 7,
+                                    "zones": {
+                                        "CH": 45,
+                                        "AS": 55
+                                    },
+                                    "executionProfile": {
+                                        "profile": "PROGRESSING_VOLUME",
+                                        "periodMs": 11,
+                                        "minionsCountProLaunchAtStart": 23,
+                                        "multiplier": 2.0,
+                                        "maxMinionsCountProLaunch": 7
+                                    }
+                                }
+                            }
+                        }
+                        """.trimIndent()
+                    )
+                )
+        ).respond(
+            response()
+                .withStatusCode(503)
```

Comment:
I would rather find interesting to simulate a 400, which is a validation issue to ensure that the validation messages returned by the server are properly logged into the Gradle console.